HomePage Update

mark-escosura opened the pull request:
feat: homepage is now a server component, moved every client component to its own folder

mark-escosura committed:
feat: homepage is now a server component, moved every client components to its own folder

mark-escosura committed:
update: changed the job interface and sampleData to align with the table created in the backend

mark-escosura committed:
add Sonner toaster from shadcn to layout and enable toast notifications for job actions

CMBVT2023 commented on src/app/job-listings/page.tsx:
Storing the job-listing element in its own folder and labeling the file page.tsx does make it an endpoint that the user can access.
This might be problematic later on if this element requires any logic that gets added to the homepage file.
Naming this file something other than page will prevent user from unintentionally accessing it.

CMBVT2023 commented on src/app/job-listings/page.tsx:
```diff
@@ -0,0 +1,155 @@
+    createdAt: new Date().toISOString(),
+    updatedAt: new Date().toISOString(),
+    tags: [
+      { id: 1, name: 'Full-time' },
```
When running `npm run build`, it produces a type error. I found that it was caused by the tag objects in the example jobs.
Their properties need to be a string so adding single quotes around id and name for all tag objects fixed the error.

CMBVT2023 commented on src/app/job-listings/page.tsx:
```diff
@@ -0,0 +1,155 @@
+    updatedAt: new Date().toISOString(),
+    tags: [
+      { id: 1, name: 'Full-time' },
+      { id: 2, name: 'Remote' },
```
Another instance where single quotes need to be added to the property names.

CMBVT2023 commented on src/app/job-listings/page.tsx:
```diff
@@ -0,0 +1,155 @@
+    createdAt: new Date().toISOString(),
+    updatedAt: new Date().toISOString(),
+    tags: [
+      { id: 3, name: 'Contract' },
```
Another instance where single quotes need to be added to the property names.

CMBVT2023 commented on src/app/job-listings/page.tsx:
```diff
@@ -0,0 +1,155 @@
+    updatedAt: new Date().toISOString(),
+    tags: [
+      { id: 3, name: 'Contract' },
+      { id: 4, name: 'Hybrid' },
```
Another instance where single quotes need to be added to the property names.

CMBVT2023 commented on src/app/job-listings/page.tsx:
```diff
@@ -0,0 +1,155 @@
+    createdAt: new Date().toISOString(),
+    updatedAt: new Date().toISOString(),
+    tags: [
+      { id: 5, name: 'Full-time' },
```
Another instance where single quotes need to be added to the property names.

CMBVT2023 commented on src/app/job-listings/page.tsx:
```diff
@@ -0,0 +1,155 @@
+    updatedAt: new Date().toISOString(),
+    tags: [
+      { id: 5, name: 'Full-time' },
+      { id: 6, name: 'Onsite' },
```
Another instance where single quotes need to be added to the property names.

CMBVT2023 commented on src/app/job-listings/page.tsx:
```diff
@@ -0,0 +1,155 @@
+      setSelectedJob(jobCards[0] ?? null);
+    }, []);
+    return (
+      <main>
```
The homepage in which this element is injected already has a main element tag.

CMWebDev02 reviewed (changes requested):
The UI elements all look good and I like the added popups when interacting with the job postings. There were a couple of things I noticed and commented on. If you need me to explain my comments further let me know. Again, the UI looks great and all user interactions are working as well.

mark-escosura committed:
Resolve conflicts

mark-escosura committed:
Merge branch 'main' into feat/job-listing-page

CMWebDev02 reviewed (dismissed):
Everything looks good.

mark-escosura committed:
feat: Enhance Job Board UI and styling, change the title and description

mark-escosura committed:
Merge branch 'feat/job-listing-page' of github.com:Bay-Valley-Tech-Code-Academy/doffy into feat/job-listing-page

mark-escosura committed:
feat: enhance UI with responsive design and search filters

- Add search and filters component with job type, location, and date filters (not working yet)
- Implement responsive mobile layout with single column view
- Add show/hide job details toggle for mobile
- Add back navigation with arrow icon
- Fix scroll area styling and spacing issues
- Make scrollbar transparent
- Add proper container classes for responsive design

ConnorBstill commented on .vscode/settings.json:
```diff
@@ -0,0 +1,5 @@
+    "cSpell.ignoreWords": [
+        "Doffy"
+    ]
+}
```
Usually we don't want to commit the .vscode folder

ConnorBstill commented on src/app/layout.tsx:
```diff
@@ -24,8 +24,8 @@ import { GeistSans } from 'geist/font/sans';
 export const metadata: Metadata = {
-  title: 'Create T3 App',
-  description: 'Generated by create-t3-app',
+  title: 'Doffy',
+  description: 'Connecting talented professionals with great companies and opportunities.',
```
This is great!

ConnorBstill commented on src/components/job-board/jobBoard.tsx:
```diff
@@ -14,73 +17,139 @@ export default function JobBoard() {
+                      <span>Posted via {job.origin}</span>
+                    </div>
+                    <div className="mt-3 flex flex-wrap gap-2">
+                      {job.tags.map((tag) => (
```
We should try to keep JavaScript logic (like `map`) outside of the main JSX return when at all possible. This helps keep the layout of the page or component easier to understand at a glance. Usually when I want to render a list of items I move the `map` to it's own function and return it and use the function in the JSX.

ConnorBstill commented on src/components/job-board/jobBoard.tsx:
```diff
@@ -14,73 +17,139 @@ export default function JobBoard() {
     );
-  }
\ No newline at end of file
+}
```
We should always have a newline at the end of every file. That's why Github puts this red icon here.
